File: code/py_rust_cpp_速度比较/count_billion.cpp
#include <iostream>
#include <chrono>

int main() {
    size_t counter = 0;
    auto start = std::chrono::high_resolution_clock::now();
    while (counter < 1000000000) {
        counter++;
    }
    auto end = std::chrono::high_resolution_clock::now();
    auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
    std::cout << "Counter: " << counter << ", Time taken: " << duration.count() << " milliseconds" << std::endl;
    return 0;
}

File: code/py_rust_cpp_速度比较/count_billion.rs
use std::time::Instant;

fn main() {
    let mut counter = 0;
    let start = Instant::now();
    
    while counter < 1_000_000_000 {
        counter += 1;
    }
    
    let end = Instant::now();
    let duration = end.duration_since(start);
    let duration_ms = duration.as_millis();
    
    println!("Counter: {}, Time taken: {} milliseconds", counter, duration_ms);
}